manifests/module_plugin.pp

```diff
@@ -89,23 +89,12 @@
     }
 
     unless $merged_conf.empty {
-      file{"${configdir}/plugin.d/${config_name}.cfg":
-        owner  => $owner,
-        group  => $group,
-        mode   => $mode,
-        ensure => $ensure
-      }
-
-      $merged_conf.each |$item, $value| {
-        ini_setting{"${name}-${config_name}-${item}":
-          ensure  => $ensure,
-          path    => "${configdir}/plugin.d/${config_name}.cfg",
-          setting => $item,
-          value   => $value,
-          require => File["${configdir}/plugin.d/${config_name}.cfg"]
-        }
-
-        Package <| tag == "mcollective_plugin_${name}_packages" |> -> Ini_setting["${name}-${config_name}-${item}"]
+      file { "${configdir}/plugin.d/${config_name}.cfg":
```

Comment:
I think this can/should use the `mcollective::config_file` define instead of repeating itself here